src/storage/admin/RebuildEdgeIndexProcessor.cpp

```diff
@@ -0,0 +1,97 @@
+        codes_.emplace_back(thriftRet);
+        onFinished();
+        return;
+    }
+
+    auto item = itemRet.value();
+    auto edgeType = item->get_schema_id().get_edge_type();
+    LOG(INFO) << "Rebuild Edge Index Space " << space << " Edge Type " << edgeType
+              << " Edge Index " << indexID;
+
+    for (PartitionID part : parts) {
+        if (req.get_is_offline()) {
+            std::unique_ptr<kvstore::KVIterator> iter;
+            auto prefix = NebulaKeyUtils::prefix(part);
+            auto ret = kvstore_->prefix(space, part, prefix, &iter);
+            if (ret != kvstore::ResultCode::SUCCEEDED) {
+                LOG(ERROR) << "Processing Part " << part << " Failed";
+                this->pushResultCode(to(ret), part);
+                onFinished();
+                return;
+            }
+
+            std::vector<kvstore::KV> data;
+            data.reserve(FLAGS_rebuild_index_batch_size);
+            int32_t batchSize = 0;
+            VertexID currentSrcVertex = -1;
+            VertexID currentDstVertex = -1;
+            while (iter && iter->valid()) {
+                if (batchSize >= FLAGS_rebuild_index_batch_size) {
+                    callingNum_ += 1;
+                    doPut(space, part, std::move(data));
+                    data.clear();
+                    batchSize = 0;
+                }
+
+                auto key = iter->key();
+                auto val = iter->val();
+                if (!NebulaKeyUtils::isEdge(key) ||
+                    NebulaKeyUtils::getEdgeType(key) != edgeType) {
+                    iter->next();
+                    continue;
+                }
+
+                auto source = NebulaKeyUtils::getSrcId(key);
+                auto destination = NebulaKeyUtils::getDstId(key);
+                if (currentSrcVertex == source || currentDstVertex == destination) {
+                    iter->next();
+                    continue;
+                } else {
+                    currentSrcVertex = source;
+                    currentDstVertex = destination;
+                }
+                auto ranking = NebulaKeyUtils::getRank(key);
+                auto reader = RowReader::getEdgePropReader(schemaMan_,
+                                                           std::move(val),
+                                                           space,
+                                                           edgeType);
+                auto values = collectIndexValues(reader.get(), item->get_fields());
+                auto indexKey = NebulaKeyUtils::edgeIndexKey(part, indexID, source,
+                                                             ranking, destination, values);
+                data.emplace_back(std::move(indexKey), "");
+                batchSize += 1;
+                iter->next();
+            }
+            doPut(space, part, std::move(data));
+        } else {
+            // TODO darion Support online rebuild index
+        }
+    }
+}
```

Comment:
Missing onFinished?

Reply:
actually, this is should not be accessed.